core/state_transition.go

```diff
@@ -543,13 +543,19 @@ func (st *StateTransition) innerTransitionDb() (*ExecutionResult, error) {
 		st.state.AddBalance(st.evm.Context.Coinbase, fee)
 	}
 
-	// Check that we are post bedrock to enable op-geth to be able to create pseudo pre-bedrock blocks (these are pre-bedrock, but don't follow l2 geth rules)
-	// Note optimismConfig will not be nil if rules.IsOptimismBedrock is true
 	// [Kroma: START]
+	baseFeeVault := params.OptimismBaseFeeRecipient
+	l1FeeVault := params.OptimismL1FeeRecipient
+	if st.evm.ChainConfig().IsPreKromaMPT(st.evm.Context.Time) {
+		baseFeeVault = params.KromaProtocolVault
+		l1FeeVault = params.KromaProposerRewardVault
+	}
+	// Check that we are post bedrock to enable op-geth to be able to create pseudo pre-bedrock blocks (these are pre-bedrock, but don't follow l2 geth rules)
+	// Note kromaConfig will not be nil if rules.IsOptimismBedrock is true
 	if kromaConfig := st.evm.ChainConfig().Kroma; kromaConfig != nil && rules.IsOptimismBedrock && !st.msg.IsDepositTx {
-		st.state.AddBalance(params.KromaProtocolVault, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.evm.Context.BaseFee))
+		st.state.AddBalance(baseFeeVault, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.evm.Context.BaseFee))
```

Comment:
Oops we need to apply this line when the block time is post-MPT, because in the case of pre-MPT, we already add base fee in L538. I think it was accidentally duplicated in previous PR to align with OP.